c10/util/BFloat16.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+// Defines the bloat16 type (brain floating-point). This representation uses
+// 1 bit for the sign, 8 bits for the exponent and 7 bits for the mantissa.
+
+#include <c10/macros/Macros.h>
+#include <iostream>
+#include <math.h>
+#include <cmath>
+#include <stdint.h>
+#include <inttypes.h>
+#include <stdio.h>
+
+namespace c10 {
+
+namespace detail {
+  inline float f32_from_bf16(uint16_t src) {
+    float res = 0;
+    uint16_t* tmp = reinterpret_cast<uint16_t*>(&res);
+    tmp[0] = 0;
+    tmp[1] = src;
+    return res;
+  }
+
+  inline uint16_t bf16_from_f32(float src) {
+    if (std::isnan(src)) {
```

Comment:
if I'm understanding the format correctly, this shouldn't be necessary.  Can you show an example where it is?

Also, does your implementation handle denormals correctly?